File: Praktikum-4/Java/PegawaiHarian.java
public class PegawaiHarian extends Pegawai {

    private final int hariKerja;

    public PegawaiHarian(String nip, String nama, double gajiPokok, int hariKerja) {
        super(nip, nama, gajiPokok);
        this.hariKerja = hariKerja;
    }

    // TODO 2: pegawai kontrak TIDAK mendapat tunjangan masa kerja.
    //         Apakah method hitungGaji() perlu di-override di sini?
    //         Pikirkan dulu, lalu tuliskan alasannya di catatan.md.

    @Override
    public double hitungGaji() {
        return super.hitungGaji() * this.hariKerja;
    }
    
    @Override
    public String jenis() {
        return "HARIAN";
    }
}
    

File: Praktikum-4/Php/PegawaiHarian.php

<?php

class PegawaiHarian extends Pegawai
{

    private int $hariKerja;

    public function __construct(string $nip, string $nama, float $gajiPokok, int $hariKerja)
    {
        parent::__construct($nip, $nama, $gajiPokok);
        $this->hariKerja = $hariKerja;
    }

    // TODO 2: pegawai kontrak TIDAK mendapat tunjangan masa kerja.
    //         Apakah method hitungGaji() perlu di-override di sini?
    //         Pikirkan dulu, lalu tuliskan alasannya di catatan.md.

    public function hitungGaji(): float
    {
        return parent::hitungGaji() * $this->hariKerja;
    }

    public function jenis(): string
    {
        return "HARIAN";
    }
}

?>
    